tasks/src/robocorp/tasks/cli.py

```diff
@@ -9,35 +9,54 @@
 code which must be automatically logged is not imported prior the the `cli.main`
 call.
 """
+
 import os
 import sys
 import warnings
+from importlib.metadata import PackageNotFoundError, version as get_version
 from typing import List, Optional, Protocol
 
+from packaging import version as version_parser
+
 
-def inject_truststore():
-    # Use certificates from native storage (if `truststore` installed)
+def _inject_truststore():
+    # Use certificates from the native storage. (if `truststore` installed and minimum
+    #  interpreter/pip versions are met)
 
-    # _RC_TEST_USE_TRUSTSTORE env var is only used to avoid unwanted warnings in tests
-    # and should not be used for other purposes
+    # NOTE(OvidiuCode): `_RC_TEST_USE_TRUSTSTORE` env var is only used to avoid
+    #  unwanted warnings under tests, and should not be used for other purposes.
     if os.getenv("_RC_TEST_USE_TRUSTSTORE", "True").lower() in ["false", "0"]:
         return
 
-    if sys.version_info >= (3, 10):
+    try:
+        pip_version = get_version("pip")
+    except PackageNotFoundError:
+        pip_ver_ok = False
+    else:
+        pip_ver_ok = version_parser.parse(pip_version) >= version_parser.parse("23.2.1")
```

Comment:
Takes pip version into account as well as observed with the legacy battletested [behaviour](https://github.com/robocorp/rpaframework/blob/master/packages/core/src/RPA/core/certificates.py#L67-L69).